### ISSUE TITLE
static.File's __init__ parameters missing documentation

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#3761]
|<img alt="jack's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">| jack reported|
|-|-|
|Trac ID|trac#3761|
|Type|defect|
|Created|2009-04-08 03:55:48Z|
|Branch|https://github.com/twisted/twisted/tree/static-file-docs-3761|

There are five or so parameters and they are currently undocumented.  This is slightly unfortunate because if you want to subclass static.File you must pass these through because the default createSimliarFile expects the subclass to have the same constructor signature.

<details><summary>Searchable metadata</summary>

```
trac-id__3761 3761
type__defect defect
reporter__jack jack
priority__normal normal
milestone__ 
branch__branches_static_file_docs_3761 branches/static-file-docs-3761
branch_author__exarkun exarkun
status__closed closed
resolution__fixed fixed
component__web web
keywords__tape tape
time__1239162948000000 1239162948000000
changetime__1239636805000000 1239636805000000
version__None None
owner__ 

```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_1"></a>|@exarkun commented|
|-|-|

(In [26666]) Branching to 'static-file-docs-3761'

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_2"></a>|@exarkun removed owner|
|-|-|

Parameter's documented.

By the way, for maximum robustness, it's very likely that any File subclass should not specifically enumerate the `__init__` parameters; there's no guarantee that we won't add new parameters or deprecated and delete old parameters.  This should happen in a backwards compatible way, so hopefully a subclass that does specifically enumerate them won't break right away, but it will need to be maintained as new Twisted releases occur.

**Comment by twisted-trac** [comment]
|[<img alt="therve's avatar" src="https://avatars.githubusercontent.com/u/159137?s=50" width="50" height="50">](https://github.com/therve)<a name="note_3"></a>|@therve set owner to @exarkun|
|-|-|

Alright, please merge.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_4"></a>|@exarkun set status to `closed`|
|-|-|

(In [26686]) Merge static-file-docs-3761

Author: exarkun
Reviewer: therve
Fixes: [#3761](https://github.com/twisted/twisted/issues/3761)

Add documentation for the parameters to `twisted.web.static.File.__init__`.

**Comment by twisted-trac** [comment]
|<img alt="Automation's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|Automation removed owner|
|-|-|
